internal/command/command.go

```diff
@@ -29,6 +29,18 @@ import (
 // It runs a command with arguments in a directory and returns the combined output as is.
 // If the command exits with a non-zero exit code, the test fails.
 func Run(ctx context.Context, dir string, args []string) (*config.TestResults, error) {
+	allCommands := true && len(args) > 1
+
+	for _, arg := range args {
+		if !strings.HasSuffix(arg, ".sh") {
+			allCommands = false
+		}
+	}
+
+	if allCommands {
```

Comment:
```suggestion
	if !slices.ContainsFunc(args, func(arg string) bool { return !strings.HasSuffix(arg, ".sh") }) {
```

Do I understand correctly that if there is any arg that does not contain `.sh` suffix, we shouldn't run each file?

You probably want to update function godoc, highlighting special case with .sh handling.

Reply:
Yes, your understanding is correct. I will add comments too.